kubetest/eks/eks.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package eks implements 'kubetest' deployer interface.
+// It uses 'awstester' and 'kubectl' binaries, rather than importing
+// EKS packages directly. This is to eliminate the need of dependency
+// management, both in upstream and downstream.
+package eks
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/aws/awstester/eksconfig"
+	"github.com/aws/awstester/eksdeployer"
+
+	"k8s.io/test-infra/kubetest/process"
+)
+
+// tester implements EKS deployer interface using "awstester" binary.
+// Satisfies "k8s.io/test-infra/kubetest/main.go" 'deployer' and 'publisher" interfaces.
+// Reference https://github.com/kubernetes/test-infra/blob/master/kubetest/main.go.
+type tester struct {
+	stopc         chan struct{}
+	cfg           *eksconfig.Config
+	awsTesterPath string
+	kubectlPath   string
+}
+
+// New creates a new EKS deployer with AWS CLI.
+func New(cfg *eksconfig.Config) (eksdeployer.Interface, error) {
+	cfg.Embedded = false
+	tr := &tester{
+		stopc: make(chan struct{}),
+		cfg:   cfg,
+	}
+
+	var err error
+	tr.awsTesterPath, err = exec.LookPath("awstester")
+	if err != nil {
+		return nil, fmt.Errorf("cannot find 'awstester' executable (%v)", err)
+	}
+	tr.kubectlPath, err = exec.LookPath("kubectl")
+	if err != nil {
+		return nil, fmt.Errorf("cannot find 'kubectl' executable (%v)", err)
+	}
+	// TODO(gyuho): replace this kubernetes native Go client
+	_, err = exec.LookPath("aws-iam-authenticator")
+	if err != nil {
+		return nil, fmt.Errorf("cannot find 'aws-iam-authenticator' executable (%v)", err)
+	}
+
+	return tr, nil
+}
+
+// Up creates a new EKS cluster.
+func (tr *tester) Up() (err error) {
+	createCmd := exec.Command(
+		tr.awsTesterPath,
+		"eks",
+		"--path="+tr.cfg.ConfigPath,
+		"create",
+		"cluster",
+	)
+	errc := make(chan error)
+	go func() {
+		ctrl := process.NewControl(
+			tr.cfg.KubetestControlTimeout,
+			time.NewTimer(tr.cfg.KubetestControlTimeout),
+			time.NewTimer(tr.cfg.KubetestControlTimeout),
+			tr.cfg.KubetestVerbose,
+		)
```

Comment:
why not pass in control from main.go and you can avoid creating new control for all the calls?